sqlite/ext/comdb2/keycomponents.c

```diff
@@ -167,7 +167,7 @@ static int systblFieldsColumn(
       break;
     }
     case STFIELD_COLNO: {
-      sqlite3_result_int64(ctx, (sqlite3_int64)pField->idx);
+      sqlite3_result_int64(ctx, (sqlite3_int64)pCur->iFieldid);
```

Comment:
If this doesn't break the existing comdb2sys tests, then I recommend adding one.